pkg/feature/api/feature.go

```diff
@@ -762,17 +761,16 @@ func (s *FeatureService) UpdateFeature(
 		if err != nil {
 			return err
 		}
-		// The feature is removed from the list to be evaluated, if it is
+		// The feature is removed from the list to be evaluated if it is
 		// archived or deleted. If it is disabled features are still evaluated.
```

Comment:
The comment is confusing to me.
It describes that it is preparing a list to be evaluated, but there is no evaluation in the updating process.

```suggestion
                 // To check if the flag to be updated is a dependency of other flags, we must validate it before updating.
		// Exclude all the archived and deleted flags from the list.
```